distributed/worker.py

```diff
@@ -2357,36 +2143,30 @@ def transition_cancelled_error(
             )
         if ts._next:
             recs[ts] = ts._next
-        return recs, smsgs
+        return recs, instructions
 
     def transition_generic_error(
         self,
         ts: TaskState,
-        exception,
-        traceback,
-        exception_text,
-        traceback_text,
+        exception: Exception,
+        traceback: object,
+        exception_text: str,
+        traceback_text: str,
         *,
         stimulus_id: str,
-    ) -> tuple[Recs, Smsgs]:
+    ) -> tuple[Recs, Instructions]:
         ts.exception = exception
         ts.traceback = traceback
         ts.exception_text = exception_text
         ts.traceback_text = traceback_text
         ts.state = "error"
-        smsg = {
-            "op": "task-erred",
-            "status": "error",
-            "key": ts.key,
-            "thread": self.threads.get(ts.key),
-            "exception": ts.exception,
-            "traceback": ts.traceback,
-            "exception_text": ts.exception_text,
-            "traceback_text": ts.traceback_text,
-        }
-
-        if ts.startstops:
-            smsg["startstops"] = ts.startstops
+        smsg = TaskErredMsg(
+            key=ts.key,
+            exception=ts.exception,
+            traceback=ts.traceback,
+            exception_text=ts.exception_text,
+            traceback_text=ts.traceback_text,
+        )
```

Comment:
Removed unused keys "status", "thread", and "startstops"

Reply:
They are not. They are subtly picked up by extensions such as TaskStream and EventStream. -__-